### PR TITLE
Fix: Handle overriding of container image in backend

### PR DESCRIPTION
**Pull request by fg91**
## Tracking issue
https://github.com/flyteorg/flyte/issues/4543

## Why are the changes needed?

The `container_image` task node override, among others, is currently broken (doesn't work with `pyflyte register` and only once with `pyflyte run`, see [tracking issue](https://github.com/flyteorg/flyte/issues/4543).

## What changes were proposed in this pull request?

Add `container_image` to `TaskNodeOverride` proto, set override in `v1alpha1.NodeSpec` in flytepropeller and actually apply the image override in the pod spec.

## How was this patch tested?

* Ran flyteadmin and flytepropeller on my dev machine. Was able to override container image in flytekit and confirm overridden image in the pod in the cluster.

   I tested this for normal python function tasks, kubeflow pytorch elastic tasks, map tasks, and array node map tasks.
* Added unit tests.

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs

https://github.com/flyteorg/flytekit/pull/2176

### CODE REVIEW

**Comment by fg91** [comment]
@hamersaw @jeevb I tagged you because I saw that both of you recently touched overriding of either extended resources or caching. Do you think I should rather tag somebody else though?

File: flyteidl/protos/flyteidl/core/workflow.proto [hunk 1]
```diff
@@ -283,4 +283,6 @@ message TaskNodeOverrides {
     // Overrides for all non-standard resources, not captured by
     // v1.ResourceRequirements, to allocate to a task.
     ExtendedResources extended_resources = 2;
+
```
Comment (pingsutw) [inline flyteidl/protos/flyteidl/core/workflow.proto:286]: could we add a comment here?

Comment (fg91) [reply flyteidl/protos/flyteidl/core/workflow.proto:300]: Of course. How about https://github.com/flyteorg/flyte/pull/4858/commits/66fe2ca2f483f9bb0d381eb938f722253fb6a8e8?

**Review by pingsutw — APPROVED** [review]
LGTM, thank you!

**Review by hamersaw — APPROVED** [review]

**Pull request merged**